### ISSUE TITLE
Color text on set of subslides

### ISSUE THREAD

**Issue by JoshuaLampert** [touying-typ/touying#111]
First of all, thank you for the very nice package. I recently built my own template based on my previous LaTeX template and managed to create almost identical results (while benefiting from typst and touying compared to LaTeX and beamer).
I mainly have two issues, which currently hinder me in completely switching to typst for presentations (this issue and an upcoming one).
I usually heavily use coloring of certain parts (in LaTeX with `\color<>[]{}`), i.e. I want some text (or part of a formula) to be colored _only in a set of subslides_. If I haven't missed anything, this is currently not supported by touying out-of-box. After looking at the source code, I figured out I can hack something together by defining it on my own. See https://typst.app/project/r2RWPmJouYepvYdkL3EtPE for an MWE. This already works quite nicely in simple situations, but doesn't work for more complex ones, where I need the callback style because IIUC I cannot get my `colors` function by `utils.methods(self)`. Note that my example works without the `semi-transparent-cover`, which I don't quite understand.
Is there interest in including such a function in touying? If yes, I would be happy to create a PR with an initial suggestion.

**Comment by OrangeX4** [comment]
This is because `semi-transparent-cover` will place paused contents into a context expression, making `touying-fn-wrapper` cannot working properly. However, you can use `config-methods(color: my color)` to add it to methods and obtains it by `#let (uncover, color) = utils.methods(self)`.

```typst
#import "@preview/touying:0.5.3": *
#import themes.simple: *

#let my_color(self: none, visible-subslides, col, cont) = {
  if utils.check-visible(self.subslide, visible-subslides) {
    text(fill: col, cont)
  } else {
    cont
  }
}

#let color(visible-subslides, col, cont) = {
  touying-fn-wrapper(
    my_color,
    last-subslide: utils.last-required-subslide(visible-subslides),
    visible-subslides,
    col,
    cont,
  )
}

#show: simple-theme.with(
  aspect-ratio: "16-9",
  config-methods(
    cover: utils.semi-transparent-cover.with(alpha: 85%),
    color: my_color,
  ),
)

= Title

== Title of Slide

#color("2-", red, "this works as I want")

#slide(repeat: 3, self => [
  #let (uncover, color) = utils.methods(self)
  This doesn't:

  $ f(x) &= (x + 1)^2 \
    uncover("2-", color("2", #red, &= x^2 + x + x + 1)) \
    uncover("3", &= x^2 + 2x + 1)
  $
])
```

**Comment by OrangeX4** [comment]
Thanks for your comments, I'm not against including such a function into touying, but I think it should be more elegant designed, for example we might use syntax.

```typst
#effect("2", text.with(fill: red))[Something]
```

**Comment by JoshuaLampert** [comment]
Ah, that makes sense! Thanks!

> Thanks for your comments, I'm not against including such a function into touying, but I think it should be more elegant designed, for example we might use syntax.

Yes, that's true. I like your suggestion. That would also make it more general. I would really like to see this feature in touying.

**Comment by OrangeX4** [comment]
I may consider it further in the future, thanks.

**Comment by JoshuaLampert** [comment]
I see a function `#effect` is included in touying:0.5.4. So thanks a lot for that @OrangeX4. However, I noticed that the callback-style doesn't seem to work with `#effect`, see my MWE here: https://typst.app/project/rBorvFignPms9v2tUqFSxz. I guess this is because `effect` is not listed here: https://github.com/touying-typ/touying/blob/e54003cb5ab69b015d4d2ec7a78d71dc8479bb78/src/configs.typ#L375-L387
Is that intentional? And if not could it be added?
I also noticed a typo in the example of the docstring for the function here: https://github.com/touying-typ/touying/blob/e54003cb5ab69b015d4d2ec7a78d71dc8479bb78/src/core.typ#L492
I guess the example should rather read `#effect("2-", text.with(fill: red))[Something]`?

**Comment by OrangeX4** [comment]
I forgot to add it to methods, I will add it in the next version. 

`#effect(text.with(fill: red), "2-")[Something]` is right, because I am considering an abbreviation by using `#let effect-red = effect.with(text.with(fill: red))`

**Comment by JoshuaLampert** [comment]
> I forgot to add it to methods, I will add it in the next version.

Great, thanks! Do you have a rough estimate, when that will be? I'm considering updating my template to touying:0.5.4. But if version 0.5.5 will be out soon, I will directly update it to 0.5.5.

> `#effect(text.with(fill: red), "2-")[Something]` is right, because I am considering an abbreviation by using `#let effect-red = effect.with(text.with(fill: red))`

I'm not sure I understand. `visible-subslides` and `fn` are positional arguments. So `#effect(text.with(fill: red), "2-")[Something]` as you suggest in the example simply errors as it is the wrong order of arguments. In the example you don't use the abbreviation `effect-red`? But also something like
```typst
#let effect-red = effect.with(text.with(fill: red))
effect-red("2-")[Something]
```
doesn't work because of the same reason. I updated the above link to demonstrate it. Maybe I'm misunderstanding what you mean. Could you clarify?

**Comment by OrangeX4** [comment]
I made a mistake, I'm sorry, I will immediately release a new version to fix.

**Comment by JoshuaLampert** [comment]
I saw your fix [f56991b](https://github.com/touying-typ/touying/commit/f56991b2a22c77c2c1c02b76abb1997149099ec9). That already looks nice. I have a small suggestion. I think the easiest is to make the suggestion in a PR. Is this fine for you @OrangeX4?

**Comment by OrangeX4** [comment]
Oh, I have already released a new waiting version. Since I have been busy recently, I just rushed to deal with these things, so it may be a little rough :-(
